### PR TITLE
Improve importer controller code readability

### PR DESCRIPTION
**Pull request by maya-r**
rename initPvcPodName to initPvcAnnotations as it initializes the scratch space annotation as well.
Make this code unconditional - the function is harmless to re-run, so we don't need a secondary check for this.
Also, we can get away with not needing to re-run the reconcile loop because we update the pvc object reference.

Reduces indentation/complexity.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
Good idea, there is one thing that might be improved

File: pkg/controller/import-controller.go [hunk 1]
```diff
@@ -324,16 +324,14 @@ func (r *ImportReconciler) reconcilePvc(pvc *corev1.PersistentVolumeClaim, log l
...
-				if err := r.initPvcAnnotations(pvc, log); err != nil {
-					return reconcile.Result{}, err
-				}
+			// Create importer pod name annotation and store in PVC
+			if err := r.initPvcAnnotations(pvc, log); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Create importer pod, make sure the PVC owns it.
+			if err := r.createImporterPod(pvc); err != nil {
```
[the remark this answers is not in the document]
Comment (maya-r) [reply pkg/controller/import-controller.go:333]: Is the new commit what you meant?

**Comment by brybacki** [comment]
tests do not pass, something does not work here

File: pkg/controller/import-controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (brybacki) [reply pkg/controller/import-controller.go:333]: I think I need to review again and rethink what I said ...

[the remark this answers is not in the document]
Comment (brybacki) [reply pkg/controller/import-controller.go:333]: this part looks good ;)

**Pull request closed without merging**